src/cpp/types/topic/WildcardTopic.cpp

```diff
@@ -40,10 +41,13 @@ bool WildcardTopic::contains(
 bool WildcardTopic::matches(
         const RealTopic& other) const
 {
-    if (utils::match_pattern(this->topic_name(), other.topic_name()))
+    if (this->topic_with_key() == other.topic_with_key())
```

Comment:
Add logic here so if this filtered topic has not set the keyed, it works for both

Reply:
:+1: